Wangscape/noise/module/ModuleFactories.cpp

```diff
@@ -0,0 +1,109 @@
+#include "ModuleFactories.h"
+#include "Gradient.h"
+#include "NormLPQ.h"
+#include "../MakeReseedable.h"
+#include <stdexcept>
+
+namespace noise
+{
+namespace module
+{
+
+Reseedable makeQuadrantSelector(Reseedable& source, bool x_positive, bool y_positive)
+{
+    if (x_positive && y_positive)
```

Comment:
The method won't work the same way; `source.translatePoint` wraps the `source` module in a `TranslatePoint` module, which adds an extra step to each `GetValue` call. If the `TranslatePoint` module doesn't do anything, then it's not necessary to wrap the module.